### ISSUE TITLE
MetricCollection with Tracker

### ISSUE THREAD

**Issue by avieini-sunbit** [PyTorchLightning/metrics#681]
## 🚀 Feature

enable the option to pass into MetricCollection the wrapper of MetricTracker including increment (per epoch) and getting the best epoch (by one of the metrics inside ) 

### Motivation
 As most of the times , I need more than one metric the first choice will be the MetricCollection but in the same way I want to have the option to keep track on my results

**Comment by SkafteNicki** [comment]
Hi @avieini-sunbit,
Could you please provide an code example of what you want to do? :]

**Comment by avieini-sunbit** [comment]
@SkafteNicki 
I think I would like to have the following option: 

```
import torch 
from torchmetrics import MetricCollection, Accuracy, Precision, Recall
target = torch.tensor([0, 2, 0, 2, 0, 1, 0, 2])
preds = torch.tensor([2, 1, 2, 0, 1, 2, 2, 2])
metric_collection = MetricTracker(MetricCollection([
    Accuracy(),
    Precision(num_classes=3, average='macro'),
    Recall(num_classes=3, average='macro')
]))
print(metric_collection(preds, target))
```
currently it's not an option as MetricTracker gets only Metric and not MetricCollection. 
I would be glad if you can add the option to include multiple metrics into the MetricTracker object and add functionality to get the best epoch (or step) per specific metric
